theme/theme.inc

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * @file
+ * This file contains all theme and preprocess functions.
+ */
+
+/**
+ * Implements template_preprocess_HOOK().
+ *
+ * @param array $variables
+ *   The theme template variables.
+ */
+function template_preprocess_islandora_openseadragon_viewer(array &$variables) {
+  module_load_include('inc', 'islandora_openseadragon', 'includes/utilities');
+  $library_path = libraries_get_path('openseadragon');
+  if (isset($variables['fedora_object']) && $variables['fedora_object'] !== NULL) {
+    $message = t("The parameter 'fedora_object' in theme('islandora_openseadragon_viewer') has been deprecated. As of 7.x-1.9, please update your code before the next release.");
+    trigger_error(check_plain($message), E_USER_DEPRECATED);
+    // Attempt to convert 'fedora_object', to 'pid'.
+    $variables['pid'] = $variables['fedora_object']->id;
+  }
+  if (isset($variables['uri']) && $variables['uri'] !== '') {
+    $message = t("The parameter 'uri' in theme('islandora_openseadragon_viewer') has been deprecated. As of 7.x-1.9, please update your code before the next release.");
+    trigger_error(check_plain($message), E_USER_DEPRECATED);
+    // Attempt to convert the 'uri' to 'tile_sources'.
+    $matches = array();
+    preg_match('/datastream\/([^\/]*)\/view/', $variables['uri'], $matches);
+    $dsid = isset($matches[1]) ? $matches[1] : FALSE;
+    $object = islandora_object_load($variables['pid']);
+    $datastream = $dsid && $object[$dsid] ? $object[$dsid] : FALSE;
+    $variables['tile_sources'] = $datastream ? array(islandora_openseadragon_tile_source($datastream)) : array();
+  }
+  $variables['id'] = $variables['viewer_id'] = 'islandora-openseadragon';
+  $variables['settings'] = array(
+    'pid' => $variables['pid'],
+    'imageServer' => variable_get('islandora_openseadragon_tilesource', 'djatoka'),
+    'djatokaServerBaseURL' => url(variable_get('islandora_openseadragon_djatoka_url', 'adore-djatoka/resolver')),
+    'iiifServerBaseURL' => url(variable_get('islandora_openseadragon_iiif_url', 'iiif')),
+    'fitToAspectRatio' => variable_get('islandora_openseadragon_fit_to_aspect_ratio', FALSE),
+    'options' => array(
+      'id' => $variables['id'],
+      'prefixUrl' => file_create_url("{$library_path}/images/"),
+      'tileSources' => $variables['tile_sources'],
+      'overlays' => islandora_openseadragon_viewer_query_solr_for_overlays($variables['pid']),
```

Comment:
Where does `islandora_openseadragon_viewer_query_solr_for_overlays`this function comes from? is this custom code of provided by other modules? can't find it anywhere. Any docs or the actual implementation to check for it?